addons/snailmail/static/src/messaging/entity/message/message.js

```diff
@@ -0,0 +1,65 @@
+odoo.define('snailmail.messaging.entity.Message', function (require) {
+'use strict';
+
+const { registerInstancePatchEntity } = require('mail.messaging.entityCore');
+
+registerInstancePatchEntity('Message', 'snailmail.messaging.entity.Message', {
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    /**
+     * Cancels the 'snailmail.letter' corresponding to this message.
+     *
+     * @returns {Deferred}
+     */
+    async cancelLetter() {
+        // the result will come from longpolling: message_notification_update
+        await this.async(() => this.env.rpc({
+            model: 'mail.message',
+            method: 'cancel_letter',
+            args: [[this.id]],
+        }));
+    },
+    /**
+     * Opens the action about 'snailmail.letter' format error.
+     */
+    async openFormatLetterAction() {
+        this.env.do_action('snailmail.snailmail_letter_format_error_action', {
+            additional_context: {
+                message_id: this.id,
+            },
+        });
+    },
+    /**
+     * Opens the action about 'snailmail.letter' missing fields.
+     */
+    async openMissingFieldsLetterAction() {
+        const letterIds = await this.env.rpc({
```

Comment:
missing `this.async()`

Reply:
But `this` is not used asynchronously here, the following code doesn't care if the entity is deleted. The action is valid as long as `letter_id` is known.

Reply:
if we add more logic in this function in the future, we could have bugs without `this.async()`. I think we should not decide not to add `this.async()` based on non-failure on current code: any async behaviour should be explicitly tagged as `async`, and deletion of entity should always prevent execution of async code on this entity, regardless of whether this is safe or not now.